src/test/encoding/readable.sh

```diff
@@ -12,85 +12,84 @@ numtests=0
 
 myversion=`./ceph-dencoder version`
 
-for arversion in `ls -v $dir/archive`
-do
-    vdir="$dir/archive/$arversion"
-#    echo $vdir
+for arversion in `ls -v $dir/archive`; do
+  vdir="$dir/archive/$arversion"
+  #echo $vdir
 
-    if [ ! -d "$vdir/objects" ]; then
-	continue;
-    fi
+  if [ ! -d "$vdir/objects" ]; then
+    continue;
+  fi
+
+  for type in `ls $vdir/objects`; do
+    if ./ceph-dencoder type $type 2>/dev/null; then
+      #echo "type $type";
+      echo "        $vdir/objects/$type"
+
+      # is there a fwd incompat change between $arversion and $version?
+      incompat=""
+      sawarversion=0
+      for iv in `ls -v $dir/archive`; do
+        if [ "$iv" = "$arversion" ]; then
+          sawarversion=1
+        fi
+
+        if [ $sawarversion -eq 1 ] && [ -e "$dir/archive/$iv/forward_incompat/$type" ]; then
+          incompat="$iv"
+        fi
+
+        if [ "$iv" = "$version" ]; then
+          break
+        fi
```

Comment:
I can't see $version in this script, is it taken from environment?

I would suggest to use 'set -u' just like 'set -e' (in other bash scripts too) to prevent running with undefined variables inside.